server/channels/app/webhook_test.go

```diff
@@ -948,7 +967,8 @@ func TestDoOutgoingWebhookRequest(t *testing.T) {
 
 	t.Run("with a large, valid response", func(t *testing.T) {
 		server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-			io.Copy(w, io.MultiReader(strings.NewReader(`{"text": "`), InfiniteReader{}, strings.NewReader(`"}`)))
+			_, err := io.Copy(w, io.MultiReader(strings.NewReader(`{"text": "`), InfiniteReader{}, strings.NewReader(`"}`)))
+			require.NoError(t, err)
```

Comment:
This error with
```
    webhook_test.go:971: 
        	Error Trace:	/mattermost/server/channels/app/webhook_test.go:971
        	            				/usr/local/go/src/net/http/server.go:2171
        	            				/usr/local/go/src/net/http/server.go:3142
        	            				/usr/local/go/src/net/http/server.go:2044
        	            				/usr/local/go/src/runtime/asm_amd64.s:1695
        	Error:      	Received unexpected error:
        	            	write tcp 127.0.0.1:44267->127.0.0.1:50410: write: connection reset by peer
        	Test:       	TestDoOutgoingWebhookRequest/with_a_large,_valid_response
```

And I think that is correct. The client writes a response that is too long and the server cuts it off.

Reply:
[`7ede219` (#29581)](https://github.com/mattermost/mattermost/pull/29581/commits/7ede219d5be9228c012e63c597d39f6406dde726) was a bit too much. My comment was only about this specific sub test.